atc/db/resource.go

```diff
@@ -333,13 +333,15 @@ func (r *resource) ResourceConfigVersionID(version atc.Version) (int, bool, erro
 	}
 
 	var id int
-	err = psql.Select("rcv.id").
-		From("resource_config_versions rcv").
-		Join("resources r ON rcv.resource_config_scope_id = r.resource_config_scope_id").
-		Where(sq.Eq{"r.id": r.ID(), "version": requestedVersion}).
-		Where(sq.NotEq{"rcv.check_order": 0}).
-		RunWith(r.conn).
-		QueryRow().
+
+	query := `
```

Comment:
It will introduce potential risk of sql injection attack since one can have a single quote in the param. This exactly why we had to rewrite the whole query block.

Reply:
You can pass parameters to `sq.Expr` without using `fmt.Printf()` by using `?` for the param. If you wanna take a look at an example, it should be similar to this https://github.com/concourse/concourse/blob/master/atc/db/resource_config_scope.go#L126.